exchange/auction.go

```diff
@@ -163,7 +160,12 @@ func (a *auction) doCache(ctx context.Context, cache prebid_cache_client.Client,
 		a.vastCacheIds = make(map[*openrtb.Bid]string, len(vastIndices))
 		for index, bid := range vastIndices {
 			if ids[index] != "" {
-				a.vastCacheIds[bid] = ids[index]
+				if competitiveExclusion && strings.HasSuffix(ids[index], hbCacheID) {
```

Comment:
I wonder if the CPU cost of `HasSuffix()` is worth the safety of verifying that the cache key is the requested cache key above.

Reply:
@hhhjort it may be. This is because if we do not validate, we may save on CPU here but the SSAI vendor would waste time fetching the bid/creative which has issues. This could result into e2e latency, which has been raised as a major concern by most clients we tried pitching this to. 

So it would be a reasonable tradeoff imo.

Reply:
Well, if we do validate, then we put in to `hb_cache_id` the cache ID that prebid cache returned, rather than the cache ID we requested to be used. This will only help the client if they are comparing all the individual `hb_cache_id`s we return and verifying that they are all indeed identical. If they don't do this, then they waste time trying to fetch a creative that does not exist.

Reply:
The point behind the HasSuffix() check is to determine which of the bids has the custom key, not to verify the correct value of the key.  Only a subset of the bids (the winning bids) has the custom key.

Reply:
If `competitiveExclusion` is on, then the only valid vast keys are the ones with the custom key. If we somehow have a key generated that is not the custom key, it will not be usable in the `competitiveExclusion` use case.